Firestore/core/src/firebase/firestore/model/database_id.h

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_MODEL_DATABASE_ID_H_
+#define FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_MODEL_DATABASE_ID_H_
+
+#include <string>
+
+namespace firebase {
+namespace firestore {
+namespace model {
+
+/** A DatabaseId represents a particular database in the Firestore. */
+class DatabaseId {
+ public:
+  /** The default name for "unset" database ID in resource names. */
+  static const char* kDefaultDatabaseId;
+
+  /**
+   * Creates and returns a new DatabaseId.
+   *
+   * @param project_id The project for the database.
+   * @param database_id The database in the project to use.
+   */
+  DatabaseId(const std::string& project_id, const std::string& database_id);
```

Comment:
These could be `absl::string_view` since we're making our own internal copy.

Reply:
done